### PR TITLE
[SPARK-25238][PYTHON] lint-python: Fix W605 warnings for pycodestyle 2.4

### PR DESCRIPTION
**Pull request by srowen**
(This change is a subset of the changes needed for the JIRA; see https://github.com/apache/spark/pull/22231)

## What changes were proposed in this pull request?

Use raw strings and simpler regex syntax consistently in Python, which also avoids warnings from pycodestyle about accidentally relying Python's non-escaping of non-reserved chars in normal strings. Also, fix a few long lines.

## How was this patch tested?

Existing tests, and some manual double-checking of the behavior of regexes in Python 2/3 to be sure.

### CODE REVIEW

File: dev/run-tests-jenkins.py [hunk 1]
```diff
@@ -115,7 +115,7 @@ def run_tests(tests_timeout):
                                          os.path.join(SPARK_HOME, 'dev', 'run-tests')]).wait()
 
     failure_note_by_errcode = {
-        1: 'executing the `dev/run-tests` script',  # error to denote run-tests script failures
+        1: 'executing the dev/run-tests script',  # error to denote run-tests script failures
```
[the remark this answers is not in the document]
Comment (HyukjinKwon) [reply dev/run-tests-jenkins.py]: Eh, I think this is a part of the Jenkins test result message. How about `# noqa` if inter complains?

File: python/pyspark/sql/streaming.py [hunk 2]
```diff
@@ -565,7 +565,7 @@ def csv(self, path, schema=None, sep=None, encoding=None, quote=None, escape=Non
             maxCharsPerColumn=None, maxMalformedLogPerPartition=None, mode=None,
             columnNameOfCorruptRecord=None, multiLine=None, charToEscapeQuoteEscaping=None,
             enforceSchema=None, emptyValue=None):
-        """Loads a CSV file stream and returns the result as a  :class:`DataFrame`.
+        r"""Loads a CSV file stream and returns the result as a  :class:`DataFrame`.
```
Comment (HyukjinKwon) [inline python/pyspark/sql/streaming.py]: tiny nit: there are two spaces within ` result as a  :class:`Da`

**Review by HyukjinKwon — APPROVED** [review]
LGTM otherwise

File: dev/create-release/generate-contributors.py [hunk 3]
```diff
@@ -88,7 +88,7 @@ def print_indented(_list):
 
 
 def is_release(commit_title):
-    return re.findall("\[release\]", commit_title.lower()) or \
+    return re.findall(r"\[release\]", commit_title.lower()) or \
```
Comment (cclauss) [inline dev/create-release/generate-contributors.py]: 1. Could we use parents to remove line terminating backslashes as recommended in PEP8?
2. Could we get rid of the use of __re__ in this instance with
```python
    return ("[release]", commit_title.lower() or
         "preparing spark release" in commit_title.lower() or
         "preparing development version" in commit_title.lower() or
         "CHANGES.txt" in commit_title)
 ```

**Review by cclauss — APPROVED** [review]
Nice work!

File: dev/create-release/generate-contributors.py [hunk 4]
```diff
@@ -67,7 +67,7 @@
 print("Release tag: %s" % RELEASE_TAG)
 print("Previous release tag: %s" % PREVIOUS_RELEASE_TAG)
 print("Number of commits in this range: %s" % len(new_commits))
-print
+print("")
```
Comment (srowen) [inline dev/create-release/generate-contributors.py]: Works in Python 2 and 3

File: dev/merge_spark_pr.py [hunk 5]
```diff
@@ -403,7 +403,7 @@ def standardize_jira_ref(text):
 
     # Extract spark component(s):
     # Look for alphanumeric chars, spaces, dashes, periods, and/or commas
-    pattern = re.compile(r'(\[[\w\s,-\.]+\])', re.IGNORECASE)
+    pattern = re.compile(r'(\[[\w\s,.-]+\])', re.IGNORECASE)
```
Comment (srowen) [inline dev/merge_spark_pr.py]: Two issues: `\.` is unnecessary, and `,-.` is probably misleading as it's a range, not three characters in the class. As it happens the range from comma to period is exactly those chars in ASCII!

**Comment by HyukjinKwon** [comment]
retest this please

**Pull request closed without merging**